File: Java/2044-count-number-of-maximum-bitwise-or-subsets.java
class Solution {
    public int countMaxOrSubsets(int[] nums) {
        Output output = new Output();
        backtrack(output, nums, 0, 0);
        return output.count;
    }
    private void backtrack(Output output, int[] nums, int current, int index) {
        if (current >= output.value) {
            if (current > output.value) {
                output.value = current;
                output.count = 1;
            }
            else {
                output.count++;
            }
        }
        for (int i = index; i < nums.length; i++) {
            backtrack(output, nums, current | nums[i], i + 1);
        }
    }
    private class Output
    {
        public int value = 0;
        public int count = 0;
    }
}

File: Python/2044-count-number-of-maximum-bitwise-or-subsets.py
class Solution:
    def __init__(self):
        self.value = 0
        self.count = 0

    def countMaxOrSubsets(self, nums: List[int]) -> int:
        self.backtrack(nums, 0, 0)
        return self.count

    def backtrack(self, nums, current, index):
        if current >= self.value:
            if current > self.value:
                self.value = current
                self.count = 1
            else:
                self.count += 1
        for i in range(index, len(nums)):
            self.backtrack(nums, current | nums[i], i + 1)
